### PR TITLE
HDDS-9853. Bump JUnit5 to 5.10.1

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Bump JUnit5 and related dependencies to current latest version.

Switch to `junit-bom` to ensure consistent versions for all JUnit5 artifacts.

https://issues.apache.org/jira/browse/HDDS-9853

## How was this patch tested?

CI:
https://github.com/adoroszlai/ozone/actions/runs/7118343498

### CODE REVIEW

**Review by hemantk-12 — COMMENTED** [review]
Thanks @adoroszlai for upgrading junit5 to latest version.

File: pom.xml [hunk 1]
```diff
@@ -225,8 +225,8 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
     <powermock2.version>2.0.4</powermock2.version>
     <jmockit.version>1.24</jmockit.version>
     <junit4.version>4.13.1</junit4.version>
-    <junit.jupiter.version>5.8.2</junit.jupiter.version>
-    <junit.platform.version>1.8.2</junit.platform.version>
+    <junit.jupiter.version>5.10.1</junit.jupiter.version>
+    <junit.platform.version>1.10.1</junit.platform.version>
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply pom.xml]: Thanks @hemantk-12 for the review, both are good points.

**Review by hemantk-12 — APPROVED** [review]
Thanks for addressing the comments.

LGTM +1.

**Comment by adoroszlai** [comment]
Thanks @hemantk-12 for the review.

**Pull request merged**